### PR TITLE
Try tests on travis, rebased with master

### PR DESCRIPTION
**Pull request by chbatey**
Replaces https://github.com/akka/akka-platform-guide/pull/72

All working now

### CODE REVIEW

File: docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -2,15 +2,22 @@
...
     <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
         <encoder>
-            <pattern>[%date{ISO8601}] [%level] [%logger] [%marker] [%thread] - %msg%n</pattern>
+            <pattern>[%date{ISO8601}] [%level] [%logger] [%X{akkaAddress}] [%marker] [%thread] - %msg%n</pattern>
         </encoder>
     </appender>
 
+    <logger name="akka.http.impl" level="WARN"/>
+    <logger name="akka.serialization" level="WARN"/>
+    <logger name="akka.cluster" level="ERROR"/>
```
Comment (patriknw) [inline docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/resources/logback-test.xml:11]: why so little logging?

**Review by patriknw — COMMENTED** [review]
what was the cause of the failures in Enno's PR? what is the difference here?

File: docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/build.sbt [hunk 2]
```diff
@@ -55,7 +55,7 @@ libraryDependencies ++= Seq(
   "com.typesafe.akka" %% "akka-persistence-testkit" % AkkaVersion % Test,
   "com.typesafe.akka" %% "akka-stream-testkit" % AkkaVersion % Test,
   "com.lightbend.akka" %% "akka-projection-testkit" % AkkaProjectionVersion % Test,
-  "org.scalatest" %% "scalatest" % "3.1.3" % Test)
+  "org.scalatest" %% "scalatest" % "3.1.2" % Test)
```
Comment (chbatey) [inline docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/build.sbt:58]: With scala 3.1.3 we don't get the failure stack trace if we start anything async in the beforeAll

File: docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/resources/logback-test.xml (hunk 1 above)
Comment (chbatey) [reply docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/resources/logback-test.xml:11]: I'll revert this change, it was just while trying to worout the failure

File: docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/scala/shopping/cart/IntegrationSpec.scala [hunk 3]
```diff
@@ -81,20 +78,20 @@ object IntegrationSpec {
...
       }
       
-      akka.actor.testkit.typed.single-expect-default = 5s
-      # For LoggingTestKit
-      akka.actor.testkit.typed.filter-leeway = 5s
-      akka.actor.testkit.typed.throw-on-shutdown-timeout = off
+      akka.actor.testkit.typed {
+        single-expect-default = 5s
+        filter-leeway = 5s
+        system-shutdown-default = 30s
```
Comment (chbatey) [inline docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/scala/shopping/cart/IntegrationSpec.scala:92]: FIXME: reduce these

**Comment by chbatey** [comment]
I spoke too soon, now we're getting timeouts later in the test. Looking into it

File: docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/scala/shopping/cart/IntegrationSpec.scala [hunk 4]
```diff
@@ -67,6 +67,8 @@ object IntegrationSpec {
       datastax-java-driver {
         basic.contact-points = ["127.0.0.1:9042"]
         basic.load-balancing-policy.local-datacenter = "datacenter1"
+        # lots of ActorSystems in the same JVM
```
Comment (patriknw) [inline docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/scala/shopping/cart/IntegrationSpec.scala:70]: thanks

File: docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/scala/shopping/cart/IntegrationSpec.scala [hunk 5]
```diff
@@ -164,18 +155,17 @@ class IntegrationSpec
   }
 
   override protected def beforeAll(): Unit = {
+    super.beforeAll()
     // avoid concurrent creation of keyspace and tables
     val timeout = 10.seconds
-    Await.result(PersistenceInit.initializeDefaultPlugins(testNode1.system, timeout), timeout)
+    PersistenceInit.initializeDefaultPlugins(testNode1.system, timeout).futureValue
```
[the remark this answers is not in the document]
Comment (chbatey) [reply docs-source/docs/modules/shopping-microservices/examples/shopping-cart-service-scala/src/test/scala/shopping/cart/IntegrationSpec.scala:163]: Okay will revert

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by chbatey** [comment]
Woo

**Pull request merged**